### PR TITLE
Load next chapter when getting to the last page

### PR DESCRIPTION
**Pull request by Forgenn**
Load next chapter when getting to the last page, in vertical reader and paged reader.

### CODE REVIEW

**Comment by AriaMoradi** [comment]
There's a problem that we have to address, settings is saved in local storage, so for users upgrading, they won't have the new keys, and when you access `settings.loadNextonEnding`, you'll get `undefined`

I think it can be solved by making sure we have all the keys at the start and set the missing ones to the default

**Comment by Forgenn** [comment]
> There's a problem that we have to address, settings is saved in local storage, so for users upgrading, they won't have the new keys, and when you access `settings.loadNextonEnding`, you'll get `undefined`
> 
> I think it can be solved by making sure we have all the keys at the start and set the missing ones to the default

Then, after retrieving the settings in Reader() in Reader.tsx, should we check if every setting has been initialized and if not, put it to the default?

**Comment by AriaMoradi** [comment]
> Then, after retrieving the settings in Reader() in Reader.tsx, should we check if every setting has been initialized and if not, put it to the default?

I'll handle it myself

File: webUI/react/src/components/ReaderNavBar.tsx [hunk 1]
```diff
@@ -182,6 +183,11 @@ export default function ReaderNavBar(props: IProps) {
             setHideOpenButton(currentScrollPos > prevScrollPos);
             setPrevScrollPos(currentScrollPos);
         }
+        if (settings.loadNextonEnding && (settings.readerType === 'ContinuesVertical' || settings.readerType === 'Webtoon')) {
+            if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
+                window.location.href = `/manga/${manga.id}/chapter/${chapter.index + 1}`;
+            }
+        }
```
Comment (AriaMoradi) [inline webUI/react/src/components/ReaderNavBar.tsx]: Each reader should provide some functionality based on the context, put this in `VerticalReader`

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

File: webUI/react/src/components/ReaderNavBar.tsx (hunk 1 above)
Comment (Forgenn) [reply webUI/react/src/components/ReaderNavBar.tsx]: I have moved this to the Vertical reader, but now the gradle check is not working even though I can compile everything locally.

Comment (AriaMoradi) [reply webUI/react/src/components/ReaderNavBar.tsx]: I made some changes, your code is not merge-able any more.

Comment (AriaMoradi) [reply webUI/react/src/components/ReaderNavBar.tsx]: > I made some changes, your code is not merge-able any more.

actually is, surprising!

File: webUI/react/src/components/reader/pager/PagedPager.tsx [hunk 2]
```diff
@@ -23,13 +23,17 @@ const useStyles = makeStyles({
...
     } = props;
 
     const classes = useStyles();
 
     function nextPage() {
-        if (curPage < pages.length - 1) { setCurPage(curPage + 1); }
+        if (curPage < pages.length - 1) {
+            setCurPage(curPage + 1);
+        } else if (settings.loadNextonEnding) {
+            window.location.href = `/manga/${manga.id}/chapter/${chapter.index + 1}`;
```
Comment (AriaMoradi) [inline webUI/react/src/components/reader/pager/PagedPager.tsx]: use this instead: https://reactrouter.com/web/api/Hooks/usehistory

File: webUI/react/src/components/reader/pager/VerticalPager.tsx [hunk 3]
```diff
@@ -20,10 +20,24 @@ const useStyles = makeStyles({
...
 
-export default function VerticalPager(props: IReaderProps) {
-    const { pages, settings, setCurPage } = props;
+export default function VerticalReader(props: IReaderProps) {
+    const {
+        pages, settings, setCurPage, manga, chapter,
+    } = props;
 
     const classes = useStyles();
+    const handleScroll = () => {
```
Comment (AriaMoradi) [inline webUI/react/src/components/reader/pager/VerticalPager.tsx]: rename to `handleLoadNextonEnding` or something similar

File: webUI/react/src/components/reader/pager/VerticalPager.tsx [hunk 4]
```diff
@@ -20,10 +20,24 @@ const useStyles = makeStyles({
...
+        if ((window.innerHeight + window.scrollY) >= document.body.offsetHeight) {
+            window.location.href = `/manga/${manga.id}/chapter/${chapter.index + 1}`;
+        }
+    };
+    useEffect(() => {
+        window.addEventListener('scroll', handleScroll);
```
Comment (AriaMoradi) [inline webUI/react/src/components/reader/pager/VerticalPager.tsx]: check for `settings.loadNextonEnding` being true, then add the event listener

**Review by AriaMoradi — CHANGES_REQUESTED** [review]

**Pull request merged**